src/EnergyPlus/HeatingCoils.cc

```diff
@@ -1360,10 +1361,16 @@ namespace HeatingCoils {
                 ShowContinueError(
                     "... if a temperature setpoint is placed at the outlet node of this heating coil, that temperature setpoint will not be used.");
                 ShowContinueError("... leaving the input field \"Temperature Setpoint Node Name\" blank will eliminate this warning.");
+                StptNodeErrorFlag = true;
             }
             MySPTestFlag(CoilNum) = false;
         }
 
+        // If the coil is temperature controlled, reset the desired outlet temperature so the temperature setpoint at the oulet node won't be used
+        if (StptNodeErrorFlag) {
+            HeatingCoil(CoilNum).DesiredOutletTemp = 0.0;
+        }
+
```

Comment:
I still don't understand why this is necessary. If the outlet node set point = -999 then the heating coil won't turn on, right? And if outlet node set point = 0 then it also won't turn on, except in winter if OA frac is large. So what's the point?

Reply:
Neal and I dug through this more and the problem as far as we can tell is that the process is running the SimulateHeatingCoilComponents function twice, once within the controlHeatingSystemToSP function and again in the calcUnitaryHeatingSystem function within the same iteration.  After the first call in the controlHeatingSystemToSP function, the PartLoadRatio gets set to 0 in [line 12855](https://github.com/NREL/EnergyPlus/blob/75fcd72b36440e1aaddf98e4b2fcd8341565340a/src/EnergyPlus/UnitarySystem.cc#L12855) since the outlet node matches the desired outlet node.  Then right before the second call for SimulateHeatingCoilComponents, QCoilReq, or HeatCoilLoad being passed in, is modified from -999 to 0 due to the PartLoadRatio being set to 0.  This is what is resulting in the coil not running in the CalcFuelHeatingCoil function due to not going into the `else if` statement on line [line 2178](https://github.com/NREL/EnergyPlus/blob/75fcd72b36440e1aaddf98e4b2fcd8341565340a/src/EnergyPlus/HeatingCoils.cc#L2178) to calculate a new TempAirOut and HeatingCoilLoad.  Our fix is to check if the `HeatCoilLoad` equals -999 or if the PartLoadRatio is 0, don't override the HeatCoilLoad to 0 and use -999 or the actual calculated PartLoadRatio.

I just pushed up the latest changes, all unit tests appear to pass.  Thank you for your feedback so far, it has helped weed through everything to figure out what is exactly happening.  Let me know your thoughts on the latest fix.